packages/Tabs/src/Tabs.js

```diff
@@ -26,73 +25,96 @@ export default function Tabs(props) {
     tabHeight,
   } = props;
 
-  const indexToUse = index !== null ? index : defaultIndex;
+  const isControlled = Boolean(onClickTab);
+
+  function getIndex() {
+    if (index === null) {
+      return defaultIndex >= 0 ? defaultIndex : null;
+    }
+    return index >= 0 ? index : null;
+  }
+
+  const indexToUse = getIndex();
   const [activeIndex, setActiveIndex] = React.useState(indexToUse);
-  const [currentFocusIndex, setFocusIndex] = React.useState(indexToUse);
-  let tabListRef = React.useRef(null);
+  const [focusIndex, setFocusIndex] = React.useState(indexToUse || 0);
+  const refList = React.useRef(null);
+  const [idTabs] = React.useState(uuidv4());
 
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
     setActiveIndex(indexToUse);
-  }, [indexToUse, setActiveIndex]);
+  }, [indexToUse]);
+
+  React.useLayoutEffect(() => {
+    const items = getItems(refList);
+    if (indexToUse !== null && items.length > 0 && isItemDisabled(items[indexToUse])) {
+      const itemIndexes = getItemIndexes(refList, { hasDisabledItems: false });
 
-  function focusAndSetIndex(index) {
-    tabListRef.querySelectorAll("[data-pka-anchor='tab'], [data-pka-anchor='tab-link']")[index].focus();
+      if (itemIndexes.length > 0) {
+        setActiveIndex(itemIndexes[0]);
+      }
+    }
+  }, []);
```

Comment:
This one is more tricky because it uses `refs` that are calculated after render. 

But, I guess if the component is controllable, this calculation will not affect the component's behaviour at all (because we'll use `index` instead of `activeIndex` instead)

Reply:
Also, I'd maybe rename `activeIndex` to make it more explicit that it is only used if the component is uncontrollable.

Reply:
Yeah, this is basically ensuring we don't set the initially selected tab to a disabled tab.  Pretty edge-casey actually and would probably benefit from a function name that describes what it's doing.